src/System.Device.Gpio/System/Device/Gpio/Drivers/Libgpiod/LibGpiodDriver.cs

```diff
@@ -1,177 +1,492 @@
+    private static string s_consumerName = Process.GetCurrentProcess().ProcessName;
+    private readonly object _pinNumberLock;
+    private readonly ConcurrentDictionary<int, LineHandle> _pinNumberToSafeLineHandle;
+    private readonly ConcurrentDictionary<int, LibGpiodV1DriverEventHandler> _pinNumberToEventHandler;
+    private readonly int _pinCount;
+    private readonly ConcurrentDictionary<int, PinValue> _pinValue;
+    private SafeChipHandle _chip;
+    private int _chipNumber;
 
-    /// <summary>
-    /// Creates an instance of the LibGpiodDriver
-    /// </summary>
-    /// <param name="gpioChip">The number of the GPIO chip to drive</param>
-    /// <remarks>
-    /// The driver version is chosen based on the installed libgpiod version on the system. To select a specific library version use the
-    /// other constructor or specify the environment variable DOTNET_IOT_LIBGPIOD_DRIVER_VERSION, see documentation
-    /// </remarks>
-    public LibGpiodDriver(int gpioChip = 0)
+    /// <inheritdoc />
+    protected internal override int PinCount => _pinCount;
+
+    // for use the bias flags we need libgpiod version 1.5 or later
+    private static bool IsLibgpiodVersion1_5orHigher()
     {
-#pragma warning disable SDGPIO0001 // Suppressing diagnostic for using experimental APIs from this same repository.
-        LibGpiodDriverFactory.VersionedLibgpiodDriver versionedLibgpiodDriver = LibGpiodDriverFactory.Instance.Create(gpioChip);
-        _driver = versionedLibgpiodDriver.LibGpiodDriver;
-        Version = versionedLibgpiodDriver.DriverVersion;
-#pragma warning restore SDGPIO0001 // Suppressing diagnostic for using experimental APIs from this same repository.
+        IntPtr libgpiodVersionPtr = LibgpiodV1.gpiod_version_string();
+        string? libgpiodVersionMatch = Marshal.PtrToStringAnsi(libgpiodVersionPtr);
+
+        if (libgpiodVersionMatch is object)
+        {
+            Version libgpiodVersion = new Version(libgpiodVersionMatch);
+            return (libgpiodVersion.Major >= 1 && libgpiodVersion.Minor >= 5);
+        }
+
+        return false;
     }
 
-    /// <summary>
-    /// Creates an instance of the LigGpiodDriver that targets specific version/s of libgpiod
-    /// <see cref="LibGpiodDriverVersion.V1"/> supports libgpiod.so.1.x to libgpiod.so.2.x
-    /// and <see cref="LibGpiodDriverVersion.V2"/> supports libgpiod.so.3.x
-    /// </summary>
-    /// <param name="gpioChip">The number of the GPIO chip to drive</param>
-    /// <param name="driverVersion">Version of the libgpiod driver to create</param>
-    /// <remarks>Alternatively, specify the environment variable DOTNET_IOT_LIBGPIOD_DRIVER_VERSION, see documentation</remarks>
-    [Experimental(DiagnosticIds.SDGPIO0001, UrlFormat = DiagnosticIds.UrlFormat)]
-    public LibGpiodDriver(int gpioChip, LibGpiodDriverVersion driverVersion)
+    private static bool s_isLibgpiodVersion1_5orHigher = IsLibgpiodVersion1_5orHigher();
+
+    private enum RequestFlag : ulong
```

Comment:
Should this be in the interop?

Reply:
It's only used here. And it's private, so no change necessary.